src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/GoogleDataprocDAO.scala

```diff
@@ -1,67 +1,211 @@
 package org.broadinstitute.dsde.workbench.leonardo.dao
 
+import java.io.{ByteArrayInputStream, File, InputStream, PrintWriter}
+import java.nio.charset.StandardCharsets
+import java.util.UUID
+
 import com.google.api.services.dataproc.model._
 import org.broadinstitute.dsde.workbench.leonardo.config.DataprocConfig
 import org.broadinstitute.dsde.workbench.google.GoogleUtilities
 import akka.actor.ActorSystem
 import akka.http.scaladsl.model.StatusCodes
+
+import scala.collection.JavaConversions._
 import com.google.api.client.auth.oauth2.Credential
 import com.google.api.client.googleapis.auth.oauth2.GoogleCredential
 import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport
 import com.google.api.client.googleapis.json.GoogleJsonResponseException
+import com.google.api.client.http.{FileContent, InputStreamContent}
 import com.google.api.client.json.jackson2.JacksonFactory
+import com.google.api.services.compute.ComputeScopes
+import com.google.api.services.compute.Compute
+import com.google.api.services.compute.model.{Firewall, Project}
+import com.google.api.services.compute.model.Firewall.Allowed
 import com.google.api.services.dataproc.Dataproc
-import com.google.api.services.pubsub.PubsubScopes
-import org.broadinstitute.dsde.workbench.leonardo.model.{ClusterRequest, ClusterResponse}
+import com.google.api.services.plus.PlusScopes
+import com.google.api.services.storage.{Storage, StorageScopes}
+import com.google.api.services.storage.model.{Bucket, ComposeRequest, StorageObject}
+import org.broadinstitute.dsde.workbench.leonardo.model.{ClusterInitValues, ClusterRequest, ClusterResponse}
 import org.broadinstitute.dsde.workbench.model.{ErrorReport, WorkbenchExceptionWithErrorReport}
 import org.broadinstitute.dsde.workbench.leonardo.errorReportSource
+
 import scala.collection.JavaConverters._
 import scala.concurrent.{ExecutionContext, Future}
 
 
-class GoogleDataprocDAO(protected val dataprocConfig: DataprocConfig)(implicit val system: ActorSystem, val executionContext: ExecutionContext)
+class GoogleDataprocDAO(protected val dc: DataprocConfig)(implicit val system: ActorSystem, val executionContext: ExecutionContext)
   extends DataprocDAO with GoogleUtilities {
 
   private val httpTransport = GoogleNetHttpTransport.newTrustedTransport
   private val jsonFactory = JacksonFactory.getDefaultInstance
-  private val cloudPlatformScopes = List(PubsubScopes.CLOUD_PLATFORM)
+  private val cloudPlatformScopes = List(ComputeScopes.CLOUD_PLATFORM)  // Maybe this would not need to be so high if we used ComputeScopes instead of Pubsub
+  private val storageScopes = Seq(StorageScopes.DEVSTORAGE_FULL_CONTROL, ComputeScopes.COMPUTE, PlusScopes.USERINFO_EMAIL, PlusScopes.USERINFO_PROFILE)
+  private val vmScopes = List(ComputeScopes.COMPUTE, ComputeScopes.CLOUD_PLATFORM)
+  private val serviceAccountPemFile = new File(dc.configFolderPath + dc.serviceAccountPemName)
```

Comment:
Minor, `new File(dc.configFolderPath, dc.serviceAccountPemName)`

It handles things like if you forget a trailing / in the parent path.